internal/grpcapi/grpcApi.go

```diff
@@ -255,6 +256,10 @@ func (api *oliveTinAPI) GetDashboardComponents(ctx ctx.Context, req *pb.GetDashb
 
 	res.AuthenticatedUser = user.Username
 
+	if res.AuthenticatedUser == "" && cfg.UrlOnUnauthenticated != "" {
+		return nil, errors.New("unauthenticated")
```

Comment:
`return nil, status.Errorf(codes.PermissionDenied, "Unauthenticated")`

Note that I'm not entirely sure if this will return a 401, or a 403.